lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -389,6 +390,7 @@ pub(crate) fn create_vector_index(
                 quantized_vectors,
                 payload_index,
                 hnsw_config: vector_hnsw_config.clone(),
+                old_indices,
```

Comment:
Is it strictly necessary to have `old_indices` in `HnswIndexOpenArgs`, it looks a bit counterintuitive. Maybe instead we could directly call `HNSWIndex::build_index` and avoid `HNSWIndex::open`?

Reply:
Resolved in #5853: now it's a part of `VectorIndexBuildArgs`, not `HnswIndexOpenArgs`.